x/ccv/provider/types/keys.go

```diff
@@ -143,30 +142,36 @@ func InitTimeoutTimestampKey(chainID string) []byte {
 	return append([]byte{InitTimeoutTimestampBytePrefix}, []byte(chainID)...)
 }
 
-// PendingCAPKey returns the key under which a pending consumer addition proposal is stored
+// PendingCAPKey returns the key under which a pending consumer addition proposal is stored.
+// The key has the following format: PendingCAPBytePrefix | timestamp | chainID
 func PendingCAPKey(timestamp time.Time, chainID string) []byte {
-	return TsAndChainIdKey(PendingCAPBytePrefix, timestamp, chainID)
-}
-
-// ParsePendingCAPKey returns the time and chain ID for a pending consumer addition proposal key
-// or an error if unparsable
-func ParsePendingCAPKey(bz []byte) (time.Time, string, error) {
-	return ParseTsAndChainIdKey(PendingCAPBytePrefix, bz)
+	timeBz := sdk.FormatTimeBytes(timestamp)
+	return AppendMany(
+		// Append the prefix
+		[]byte{PendingCAPBytePrefix},
+		// Append the time bytes
+		timeBz,
```

Comment:
These bytes are not of set length, correct? If that's the case, it should be at the end of the key, or hashed to a known byte length. Also, does `TsAndChainIdKey` still exist?

Reply:
you could put the hashed chainID before the time bytes

Reply:
> If that's the case, it should be at the end of the key, or hashed to a known byte length. 

Why? The idea is to oder by timestamp when iterating over the proposals. 

> Also, does TsAndChainIdKey still exist?

Yes, but that prefixes the timestamp with its length, and then the iteration will be in the order of the time length. 

> you could put the hashed chainID before the time bytes

But then you cannot break the iteration based on the timestamp as you iterate in the order of chainIDs.

Reply:
On bytes not being a set length, and why that's a problem, see https://github.com/cosmos/interchain-security/issues/18

Reply:
There's multiple solutions, like removing chainID from the key altogether and storing an array of proposals for each time step

Reply:
Separators could also work